src/action/common/configure_shell_profile.rs

```diff
@@ -1,51 +1,142 @@
+under which Fish will look for a file named
+[`PROFILE_FISH_SUFFIX`].
+*/
+const PROFILE_FISH_PREFIXES: &[&str] = &[
+    "/etc/fish",              // standard
+    "/usr/local/etc/fish",    // their installer .pkg for macOS
+    "/opt/homebrew/etc/fish", // homebrew
+    "/opt/local/etc/fish",    // macports
+];
 const PROFILE_TARGETS: &[&str] = &[
     "/etc/bashrc",
     "/etc/profile.d/nix.sh",
     "/etc/zshrc",
     "/etc/bash.bashrc",
     "/etc/zsh/zshrc",
-    // TODO(@hoverbear): FIsh
 ];
-const PROFILE_NIX_FILE: &str = "/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh";
+const PROFILE_NIX_FILE_SHELL: &str = "/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh";
+const PROFILE_NIX_FILE_FISH: &str = "/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.fish";
 
 /**
 Configure any detected shell profiles to include Nix support
  */
 #[derive(Debug, serde::Deserialize, serde::Serialize, Clone)]
 pub struct ConfigureShellProfile {
+    create_directories: Vec<StatefulAction<CreateDirectory>>,
     create_or_append_files: Vec<StatefulAction<CreateOrAppendFile>>,
 }
 
 impl ConfigureShellProfile {
     #[tracing::instrument(skip_all)]
     pub async fn plan() -> Result<StatefulAction<Self>, ActionError> {
         let mut create_or_append_files = Vec::default();
+        let mut create_directories = Vec::default();
+
+        let shell_buf = format!(
+            "\n\
+            # Nix\n\
+            if [ -e '{PROFILE_NIX_FILE_SHELL}' ]; then\n\
+            {inde}. '{PROFILE_NIX_FILE_SHELL}'\n\
+            fi\n\
+            # End Nix\n
+        \n",
+            inde = "    ", // indent
+        );
+
         for profile_target in PROFILE_TARGETS {
-            let path = Path::new(profile_target);
-            if !path.exists() {
-                tracing::trace!("Did not plan to edit `{profile_target}` as it does not exist.");
+            let profile_target_path = Path::new(profile_target);
+            if let Some(parent) = profile_target_path.parent() {
+                if !parent.exists() {
+                    tracing::trace!(
+                        "Did not plan to edit `{profile_target}` as it's parent folder does not exist."
+                    );
+                    continue;
+                }
+                create_or_append_files.push(
+                    CreateOrAppendFile::plan(
+                        profile_target_path,
+                        None,
+                        None,
+                        0o0755,
+                        shell_buf.to_string(),
+                    )
+                    .await?,
+                );
+            }
+        }
+
+        let fish_buf = format!(
+            "\n\
+            # Nix\n\
+            if test -e '{PROFILE_NIX_FILE_FISH}'\n\
+            {inde}. '{PROFILE_NIX_FILE_FISH}'\n\
+            end\n\
+            # End Nix\n\
+        \n",
+            inde = "    ", // indent
+        );
+
+        for fish_prefix in PROFILE_FISH_PREFIXES {
```

Comment:
Probably won't matter until something, somewhere changes, but I wonder if we should track if we've found a prefix that exists, and otherwise warn that we didn't find any, and so the Nix install will be dysfunctional?

Reply:
Hmm a prefix not existing implies fish isn't installed, and we don't need to provision support. (At least, this is what the current nix-isntaller does)